src/lib/object.ts

```diff
@@ -4,7 +4,43 @@ import { UnexpectedError } from '../internal/errors';
  * Creates a deep clone of the value.
  */
 export function clone<T>(value: T): T {
-  return JSON.parse(JSON.stringify(value)) as T;
+  if (value === null) {
+    return value;
+  }
+
+  if (value instanceof Date) {
+    return new Date(value.getTime()) as unknown as T;
+  }
+
+  if (value instanceof Array) {
+    const arrayCopy = [] as unknown[];
+    value.forEach(item => {
+      arrayCopy.push(item);
+    });
+
+    return arrayCopy.map((item: unknown) =>
+      clone<unknown>(item)
+    ) as unknown as T;
+  }
+
+  if (Buffer.isBuffer(value)) {
+    return Buffer.from(value) as unknown as T;
+  }
+
+  if (typeof value === 'object' && value !== {}) {
```

Comment:
What is the intent with this condition: `value !== {}`? It never evaluates to false since you are comparing references. If you want to check for an empty object, you will need something like `Object.keys(value).length > 0`

Reply:
However I would dropping this condition altogether, I think it's an unnecessary optimization, the code will still work with empty objects (though make sure to test it).

Reply:
@jnv you are right the condition for empty object was wrong. I removed it and added test.